src/transformers/models/wav2vec2/processing_wav2vec2.py

```diff
@@ -0,0 +1,142 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Speech processor class for Wav2Vec2
+"""
+from contextlib import contextmanager
+
+from .feature_extraction_wav2vec2 import Wav2Vec2FeatureExtractor
+from .tokenization_wav2vec2 import Wav2Vec2CTCTokenizer
+
+
+class Wav2Vec2Processor:
```

Comment:
Should this have a parent class?

Reply:
Actually, it looks like this could be a very generic class and doesn't need to be Wav2Vec2 specific

Reply:
Okey, I've been discussing quite a bit @patil-suraj about this. 
I do think that the functionality of the class will always be the same so it doesn't need to be Wav2Vec2 specific.
On the other hand, the docstring and the type checking is definitely very much different for each class.

I think there are three options:

a) Have only a single `Preprocessor` class that handles everything and only use this class for all models. The huge downside to me here is that it won't really be possible to have nice docstrings (linking to the correct feature extractor, tokenizer), and the type checking is not really possible. Also it's not really in line IMO with the general lib philosophy of having one level of abstraction model customized classes that inherit from a `Pretrained...` class. I guess `EncoderDecoderModel` is an exception, but it's definitely a downside for `EncoderDecoderModel` to me. Also there might be specific functionality that is different for some models, that we don't see yet.

b) Also make some kind of `PretrainedProcessor` that holds most of the functionality (even though there really isn't much) and overwrite the docstrings & type checking. For this we would still give the name `Wav2Vec2Processor`.

c) Leave it as it is.

=> taking this all into account I tend to b) & c) and since it's the first model of it's kind I'd just go for c) and leave it as it is now because we could always refactor it to b) without breaking change. I'm not really in favor of a) because of the reasons above, but maybe there is a huge advantage to a) I'm not seeing...What do you think @LysandreJik @sgugger ?

Reply:
I'm okay with option c)!